custom_components/econet300/entity.py

```diff
@@ -112,11 +112,16 @@ async def async_added_to_hass(self):
         # Retrieve the value from sysParams or regParams  or paramsEdits
         value = (
             sys_params.get(expected_key)
-            if sys_params.get(expected_key) is not None
+            if sys_params is not None and sys_params.get(expected_key) is not None
```

Comment:
Did you check this logic? Because as i thought that your earlyer created if/else logic would be enough